handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -308,7 +308,8 @@ private OpenSslSession wrapSessionIfNeeded(OpenSslSession session) {
     final synchronized void setupHandshakeSession() {
         OpenSslSession old = session;
         if (old.isNullSession()) {
-            session = wrapSessionIfNeeded(newOpenSslSession(SSL.getSession(ssl)));
+            // We are still in the handshake phase so always use -1 as there is "no underlying" session yet.
+            session = wrapSessionIfNeeded(newOpenSslSession(-1));
```

Comment:
Can we use an `OpenSslNullSession` in this case instead of `-1`?

Reply:
nope as we already have some stuff we pull from the `SSL` object.